lib/Doctrine/Common/DataFixtures/Executor/ORMExecutor.php

```diff
@@ -45,6 +48,8 @@ public function __construct(EntityManagerInterface $em, ?ORMPurgerInterface $pur
             $this->purger->setEntityManager($em);
         }
 
+        $this->executorStrategy = $executorStrategy ? new $executorStrategy($this) : new SingleTransactionStrategy($this);
```

Comment:
This is a circular dependency. A better design might be the following:

```php
<?php
interface Executor
{
    abstract public function execute(array $fixtures, $append = false);
}

abstract class AbstractExecutor implements Executor
{
}

class ORMExecutor extends AbstractExecutor
{
}

class MultipleTransactionORMExecutor implements Executor
{
   public function __construct(private readonly ORMExecutor $executor)
   {}


   public function execute(array $fixtures, $append = false)
   {
   }
}
```

What do you think?

Reply:
It looks good too, I updated the PR